src/components/header/Header.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { Link, withRouter } from 'react-router-dom';
+import classes from './Header.module.scss';
+import { AuthContext } from '../../store/contexts/AuthContext';
+import { FavoriteContext } from '../../store/contexts/FavoriteContext';
+import { youtubeApi, youtubeApiQuotaExceeded } from '../../services/YouTubeService';
+import icon from '../../assets/icon.png';
+import { useDebounce } from '../../hooks/useDebounce';
+
+const Header = (props) => {
+  // eslint-disable-next-line
+  const [favState, favDispatch] = useContext(FavoriteContext);
+  const [state, dispatch] = useContext(AuthContext);
+  const [searchTerm, setSearchTerm] = useState('wizeline');
+
+  const debouncedSearchTerm = useDebounce(searchTerm, 500);
+
+  const createReferenceMap = (videos = []) => {
+    const map = new Map();
+    videos.forEach((video, index) => {
+      map.set(video.id, index);
+    });
+    return map;
+  };
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const response = await youtubeApi.get('/search', {
+          params: {
+            q: searchTerm,
+          },
+        });
+
+        // TODO: Using when quota exceeded
+        // const response = await youtubeApiQuotaExceeded(searchTerm);
+
+        const {
+          data: { items },
+        } = response;
+
```

Comment:
This can be refactored into a custom hook, something like useSearch, this way you keep things unmixed, most of the code in this file has to do with data fetching, which is unrelated to the component

Reply:
Awesome, thanks for the feedback.